source/virtualBuffers/__init__.py

```diff
@@ -218,6 +218,37 @@ def _getTextRange(self,start,end):
 			return u""
 		return NVDAHelper.VBuf_getTextInRange(self.obj.VBufHandle,start,end,False) or u""
 
+	def _getPlaceholderAttribute(self, attrs, placeholderAttrsKey):
+		"""Gets the placeholder attribute to be used.
+				@returns The placeholder attribute when there a value is not present in the related field.
+					None when the field has a value
+		"""
+		placeholder = attrs.get(placeholderAttrsKey)
+		# For efficiency, only check if it is valid to return placeholder when we have a placeholder value to return.
+		if not placeholder:
+			return None;
+		# Get the start and end offsets for the value. This can be used to check if the field has a value.
+		try:
+			start, end = self._getOffsetsFromFieldIdentifier(
+				int(attrs.get('controlIdentifier_docHandle')),
+				int(attrs.get('controlIdentifier_ID')))
+		except (LookupError, ValueError):
+			log.debugWarning("unable to get offsets used to fetch content")
+			return placeholder
+		else:
+			valueLen = end - start
+			if not valueLen: # value is empty, use placeholder
+				return placeholder
+			# Because fetching the value of the field could result in a large amount of text
+			# we only do it in order to check for space, or newline.
+			# We first compare the length by comparing the offselts, if the length is less than 2 (ie
+			# could hold space or \n)
+			if valueLen < 2:
+				controlFieldText = self.obj.makeTextInfo(textInfos.offsets.Offsets(start, end)).text
+				if not controlFieldText or controlFieldText == ' ' or controlFieldText == '\n':

```

Comment:
Not actually 100% sure that we need to have exceptions for space and newline here.